RadixWalletTests/Features/AssetTransferTests/AssetTransferDepositRuleTests.swift

```diff
@@ -0,0 +1,143 @@
+import EngineToolkit
+@testable import Radix_Wallet_Dev
+import XCTest
+
+@MainActor
+final class AssetTransferDepositRuleTests: TestCase {
+	static let recipientAccount = Profile.Network.Account.testValueIdx0
+	static let resourceAddress = try! ResourceAddress(validatingAddress: "resource_rdx1tkk83magp3gjyxrpskfsqwkg4g949rmcjee4tu2xmw93ltw2cz94sq")
+
+	static let onLedgerAccountWithResource = OnLedgerEntity.Account(
+		address: recipientAccount.address,
+		atLedgerState: .init(version: 0, epoch: 0),
+		metadata: .init(nil),
+		fungibleResources: .init(
+			xrdResource: .init(
+				resourceAddress: resourceAddress,
+				atLedgerState: .init(version: 0, epoch: 0),
+				amount: .one(),
+				metadata: .init(nil)
+			)
+		),
+		nonFungibleResources: [],
+		poolUnitResources: .init(radixNetworkStakes: [], poolUnits: [])
+	)
+
+	// MARK: - AcceptAll deposit rule
+	func test_GIVEN_depositRuleIsAcceptAll_WHEN_resourceHasNoException_THEN_noSignatureIsRequired() async {
```

Comment:
minor, but I highlight the GIVEN WHEN THEN in test names by used of `__` instead of `_` thus making tests easier to read.

Also annotating `// GIVEN` `// WHEN` and `// THEN` in the body of the test, *matching* the method name, greatly increases readability!

Reply:
It would be nice to use such tools as https://github.com/Quick/Quick to do this kind of testing.